src/model/config.ts

```diff
@@ -0,0 +1,54 @@
+import BN from "bn.js";
+import { ConfigurableCorePool } from "../core/configurablePool";
+
+export class PoolConfig {
+    id: string;
+    tickSpacing: number;
+    token0: string;
+    token1: string;
+    fee: number;
+}
+
+export class Record {
+    id: string;
+    eventId: number;
+    actionName: string;
```

Comment:
Considering the types of actions are limited and not arbitrary, maybe it's safer and more clear if using a typed `enum` e.g. `ActionType` here and updating the variable name to `actionType`.

Reply:
I agree.